console/src/main/java/org/georchestra/console/filters/UserChangeFilter.java

```diff
@@ -0,0 +1,39 @@
+package org.georchestra.console.filters;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class UserChangeFilter implements Filter {
```

Comment:
2 remarks on the current implementation choices:

* you are using plain `javax.servlet` Filter, as we are using spring / spring security, why not integrating a spring filter to the spring security filter chain instead ? (https://docs.spring.io/spring-security/reference/servlet/architecture.html#adding-custom-filter). Also considering the fact that you are actually using spring security objects in the code (SecurityContextHolder on line 22), it would make more sense to me to stay in a "pure spring" approach. It will also make any move to jakartaEE easier, as we are only relying on the framework which should abstract the `javax.*` / `jakarta.*` upgrade / renaming.
* If the idea is to be able to reuse the same filter across different georchestra webapps, I'd nest the code into another module, like commons or spring-security-proxy-integration

Reply:
I moved it to commons but didn't succeed to make it working in `applicationContext-security.xml`. 

App didn't react as expected..